### ISSUE TITLE
Uglify and mangle: true

### ISSUE THREAD

**Issue by webnoob** [jpex-js/vue-inject#15]
I've just updated my production alpha site and had an issue with the dependencies not being resolved. I immediately thought of the mangle option and it's working fine with mangling disabled.

Is there any way to support mangling?

**Comment by jackmellis** [comment]
What kind of mangling are we talking? Is this the option that renames object properties? If so (and if this is the reason why it's not working), then I don't think it's possible to support it with vue-inject as it relies heavily on dynamically-registered properties and runtime property accessing

**Comment by webnoob** [comment]
Yeah, it renames. I thought it would be the case that nothing could be done but thought it worth checking.

**Comment by webnoob** [comment]
I thought it worth coming back to this as I've been looking at [Awilix](https://www.npmjs.com/package/awilix#installation). They have a method whereby the DI resolution isn't done on constructor parameters but is instead done using a proxy (look for `InjectionMode` in the docs)

Is this something that you would think about implementing in `vue-injext`?
